.goreleaser.yml

```diff
@@ -7,8 +7,8 @@ builds:
       - darwin
       - linux
     goarch:
-      # sops is only available for amd64
       - amd64
+      - arm64
```

Comment:
this looks good so far...

Is there something we need to document about the required sops version? I think we can specify a >= version in the brewfile, deb, etc.

Reply:
I've been looking at GoRelease docs around [Homebrew Taps](https://goreleaser.com/customization/homebrew/) but it doesn't give much info on how to specify the version for a dependency.  I wonder if I can use [targets](https://goreleaser.com/customization/build/#build-hooks) to see about specifying the builds.  I wonder if we could split releases up how other upstream repositories do (e.g. `sopstool-<VERSION>.OS.arch`)

```
sopstool-0.5.0.darwin.arm64
```

I'm not sure if someone installing the tool will run into issues if we change the release name structure.  I'll keep reading in GoReleaser and see what we can do.

Reply:
The Brewfile seems to be only for sopstool dev vs sopstool consumption.

@physik932 You should check if the [`sopsinstall.sh`](https://github.com/Ibotta/sopstool/blob/master/sopsinstall.sh) file installs the ARM version properly. If not, it'll need updated; ditto with [`sopstoolinstall.sh`](https://github.com/Ibotta/sopstool/blob/master/sopstoolinstall.sh).

Although, those are auto-generated by `godownloader` which is deprecated: https://github.com/goreleaser/godownloader/issues/207. 🤔

Reply:
I'm looking through some of the options here https://goreleaser.com/customization/archive/ et al, and wondering about if we're going to have to make a breaking change in the artifact formats. we made the simplification/mistake of leaving arch out of the file paths.  I was hoping we could 'do it twice' by archiving x86_64 and 'blank' but I'm not seeing an easy way to do this. it may be custom post-build action.

Reply:
we'll also want to enable docker manifests and build an arm docker.

Reply:
I am poking around popular upstream GitHub repositories written in golang (helm, helmfile, sops), and they all use a Makefile and then some package manager

- [helm](https://github.com/helm/helm/blob/main/Makefile#L5) uses [gox](https://github.com/mitchellh/gox#usage) with the target arch and OS set up as variables in their [Makefile](https://github.com/helm/helm/blob/main/Makefile#L4-L5)
- sops uses [fpm](https://github.com/jordansissel/fpm) in their [Makefile](https://github.com/mozilla/sops/blob/master/Makefile#L51-L92) invoked by a github action

I'm not saying we should switch necessarily, I'm just poking around to see implementations and what might be easiest.  I'm gonna dig into those Archive docs for GoReleaser a bit to see what it would look like.

As far as a breaking change, it might be necessary if it means setting up for future success with OS and architecture versions?

Reply:
goreleaser definitely supports all this just fine. it builds on my machine. But the archive formats we've previously defined don't. I think we'll have to update those formats.